deployment/helm/values.yaml

```diff
@@ -31,8 +31,9 @@ controller:
     tag:
     # pullPolicy is the Akri Controller pull policy
     pullPolicy: Always
-  # onlyOnMaster dictates whether the Akri Controller will only run on a master node
-  onlyOnMaster: true
+  # onlyOnMaster dictates whether the Akri Controller will only run on node with 
+  # the master node label 'node-role.kubernetes.io/master'
```

Comment:
Should we clarify here the value of the label too, since the actual filter is not just on the label key?

Reply:
Can you clarify "since the actual filter is not just on the label key"? Isnt this the case? OnlyOnMaster is specifying that the controller will only run on a node with that label, so it seems to be a filter on that label key. Are you suggesting we add more commenting here about what the label means?

Reply:
I think removing the node selector means it can run anywhere...maybe the master, maybe not?

Reply:
My point was that the filter is this:
```
nodeSelector:
        node-role.kubernetes.io/master: ""
```

So it is not just label key, it is both key and value (empty string). As such, my argument was that the document should fully state what is the nodeSelector (key+value). Or do you not agree?

Reply:
we could create a more generalized approach to onlyOnMaster.  something like we do for imagePullSecrets.  that would allow someone to specify whatever they wanted (and it could default to nothing):

```
# nodeSelector is the array of NodeSelectors that help target nodes with specific labels.
# This can be set from the helm command line using `--set nodeSelectors[0].name-role.kubernetes.io/master=""`
nodeSelectors: []
```

Reply:
one thing i just noticed is that it looks like nodeSelector is being specified twice in controller.yaml (once for onlyLinux and once for onlyOnMaster).  i just tested it and that doesn't work ... only the last thing set ends up in the kubernetes object.

Reply:
to fix the duplicate nodeSelector issue (and add more configurations for someone), we could adjust:

values.yaml:
```yaml
  # onlyOnMaster dictates whether the Akri Controller will only run on a node labeled with `node-role.kubernetes.io/master: ""`
  onlyOnMaster: false
  # linuxOnly dictates whether the Akri Controller will only run on a linux node
  linuxOnly: true
  # nodeSelectors is the array of nodeSelectors used to target nodes for the Akri Controller to run on
  # This can be set from the helm command line using `--set nodeSelectors.label="value"`
  nodeSelectors: {}
```
controller.yaml (remove both existing nodeSelector lines and replace with) :
```yaml
      {{- if or .Values.controller.linuxOnly .Values.controller.onlyOnMaster }}
      nodeSelector:
        {{- if .Values.controller.nodeSelectors }}
          {{- toYaml .Values.controller.nodeSelectors | nindent 8 }}
        {{- end }}
        {{- if .Values.controller.linuxOnly }}
        "kubernetes.io/os": linux
        {{- end }}
        {{- if .Values.controller.onlyOnMaster }}
        node-role.kubernetes.io/master: ""
        {{- end }}
      {{- else }}
        {{- if .Values.controller.nodeSelectors }}
      nodeSelector:
          {{- toYaml .Values.controller.nodeSelectors | nindent 8 }}
        {{- end }}
      {{- end }}
```

then these commands would all work:
```sh
# keeps linux only, sets only on master, lets user add any nodeSelector they want
helm template ./deployment/helm --set controller.linuxOnly=true,controller.nodeSelectors.foo=bard,controller.onlyOnMaster=true
# defaults to only on linux and whatever the default is for onlyOnMaster
helm template ./deployment/helm
```